### ISSUE TITLE
Pluggable graph database to support Janus

### ISSUE THREAD

**Issue by djcoder100** [amundsen-io/amundsen#160]
Is it possible to plugin in another graph implementation like Janus https://janusgraph.org/ as the meta data catalogue or is the architecture tied to Neo4j

Reasons for this is we have Janus incorporated already into our stack and wanted to stick one graph database for operational support.

**Comment by jinhyukchang** [comment]
Hey @djcoder100 

Is there any chance that Janus support Cypher? If that is the case, it would be doable project.
If not, both metadata service and databuilder need to support Janus which is actually fine per architecture perspective but it just needs some work.

Thanks!
Jin

**Comment by feng-tao** [comment]
cc @verdan , also Atlas uses janus as the backend which ING team is leading the integration between Amundsen and Atlas the backend.

**Comment by feng-tao** [comment]
but as @jinhyukchang mentioned, you could build a proxy layer like the other one(neo4j, atlas proxy) in the metadata repo to support the integration.

**Comment by jornh** [comment]
Related thread on slack: https://amundsenworkspace.slack.com/archives/CGFBVT23V/p1570561231404600

**Comment by quondammelody** [comment]
tl;dr from the thread, yes! We (Square) are planning to add a gremlin proxy in the near future (maybe in a few weeks?) and open source that, so it should be totally compatible with janusgraph (or Neptune, etc).

**Comment by friendtocephalopods** [comment]
tl;dr from the thread, yes! We (Square) are planning to add a gremlin proxy in the near future (maybe in a few weeks?) and open source that, so it should be totally compatible with janusgraph (or Neptune, etc).

**Comment by djcoder100** [comment]
> tl;dr from the thread, yes! We (Square) are planning to add a gremlin proxy in the near future (maybe in a few weeks?) and open source that, so it should be totally compatible with janusgraph (or Neptune, etc).

That would be great to see a short write up as Janus is what we would like to integrate as then we can standup in gcp cloud using big table as a storage layer for scale.

**Comment by hagridaaron** [comment]
@friendtocephalopods I'm wondering about that gremlin proxy - would that allow me to use something like AWS Neptune as opposed to Neo4j?

**Comment by friendtocephalopods** [comment]
That's right! We're actually working through a neptune flavor of the gremlin proxy right now, hoping to be switched over within a few weeks. We'd be happy to upstream that when it's ready.

**Comment by feng-tao** [comment]
this ticket is duplicated by https://github.com/amundsen-io/amundsengremlin
